models/result.go

```diff
@@ -49,7 +55,49 @@ func (r *Result) createEvent(status string, details interface{}) (*Event, error)
 		if err != nil {
 			return nil, err
 		}
-		e.Details = string(dj)
+
+		if EVENT_DATA_SUBMIT == status && c.PublicKeyId != 0 { // Zero is unset
```

Comment:
Minor nit - let's switch this around
```suggestion
		if status == EVENT_DATA_SUBMIT && c.PublicKeyId != 0 { // Zero is unset
```